x-pack/docs/en/security/securing-communications/security-basic-setup.asciidoc

```diff
@@ -0,0 +1,277 @@
+authentication. Applying TLS ensures that a malicious node cannot join the
+cluster and exchange data with other nodes. While implementing username and
+password authentication at the HTTP layer is useful for securing a local
+cluster, there is no internode security without configuring TLS.
+
+Configuring TLS is the basic security setup to prevent unauthorized nodes from
+accessing to your cluster.
+
+[[basic-setup-prerequisites]]
+==== Prerequisites
+
+Complete all steps in <<security-minimal-setup,Minimal security for the Elastic Stack>> on every node in your cluster. After enabling {es} security and setting
+passwords for the built-in users, you can encrypt communications between your
+nodes with TLS.
+
+[[generate-certificates]]
+==== Generate the certificate authority
+
+You can add as many nodes as you want in a cluster but they must be able to communicate with each other. The communication between nodes in a cluster is handled by the transport module. To secure your cluster, you must ensure that the internode communications are encrypted and that mutual TLS is configured.
+
+In a secured cluster, {es} nodes use certificates to identify
+themselves when communicating with other nodes.
+
+The cluster must validate the authenticity of these certificates. The
+recommended approach is to trust a specific certificate authority (CA). When
+nodes are added to your cluster they can use a certificate signed by the same
+CA.
+
+For the transport layer, we recommend using a separate, dedicated CA instead
+of an existing, possibly shared CA. Use the `elasticsearch-certutil` tool to
+generate a CA for your cluster.
+
+NOTE: If you're running a multi-node cluster, jump to
+<<multi-node-certs,Generate certificates for a multi-node cluster>> and use
+that method of generating a CA and certificates.
+
+. Use the `elasticsearch-certutil` tool to generate a CA for your cluster.
++
+[source,shell]
+----
+./bin/elasticsearch-certutil ca
+----
+
+   a. When prompted, accept the default file name, which is `elastic-stack-ca.p12`. This file contains the public certificate for your CA and the private key used to sign certificates for each node.
+
+   b. Enter a password for your CA. You can leave the password
+   blank if you're configuring security for a minimal development environment.
+
+. Generate a certificate and private key for your node. You include the
+   `elastic-stack-ca.p12` output file that you generated in the previous step.
++
+[source,shell]
+----
+./bin/elasticsearch-certutil cert --ca elastic-stack-ca.p12
+----
+
+   a. Enter the password for your CA.
+
+   b. Create a password for the certificate.
++
+The output file is a keystore named `elastic-certificates.p12`. This file
+contains a node certificate, node key, and CA certificate.
+
+. Copy the `elastic-certificates.p12` file to the `ES_PATH_CONF`
+   directory.
+
+*Next*: <<encrypt-internode-communication>>
+
+[[multi-node-certs]]
```

Comment:
I'm not sure what is the purpose of this distinction here. In both cases we then suggest to use `xpack.security.transport.ssl.verification_mode: certificate` in the settings, so what is the added benefit of using different key/certificate for each node ?  Since hosname verification is not happening, leaking any of this keypairs would allow an attacker to join a node to the cluster (assuming network connectivity ) ,isn't it just simpler to use one ?

Also we talk about node_s_ everywhere already. Why would someone want to enable transport layer TLS for a single node cluster ( with transport client deprecated and being removed soon ) ?

Reply:
@jkakavas, I was trying to distinguish between a single and multi-node cluster. Are you suggesting that we *not* encourage users to follow this method for [generating certificates](https://elasticsearch_68946.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/security-basic-setup.html#multi-node-certs)? I was under the assumption that we want users to generate a different certificate for each node. Are you suggesting that they create only a single `<node-name>.p12` certificate file, and then copy that file to every node in the cluster?

Reply:
I think @jkakavas's point is that the main (only?) value of having a different certificate per node is that it allows you set `verification_mode: full` which may offer some additional security.

Advising people to generate a cert per node, and then set  `xpack.security.transport.ssl.verification_mode: certificate` gives them all the pain of having to manage multiple certificates but none of the benefits.

I think the answer is that we should be setting
```
xpack.security.transport.ssl.verification_mode: full
```
in the multi-certificate case (or else remove the multi-node-cert option entirely).